packages/firebase_ml/android/src/main/java/io/flutter/plugins/firebaseml/FirebaseMLPlugin.java

```diff
@@ -40,21 +50,86 @@ public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBindin
     channel.setMethodCallHandler(this);
   }
 
+  @RequiresApi(api = Build.VERSION_CODES.N)
   @Override
-  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
+  public void onMethodCall(@NonNull MethodCall call, @NonNull final Result result) {
+    final FirebaseCustomRemoteModel remoteModel;
+    String modelName;
+
     switch (call.method) {
-      case "downloadRemoteModel":
-        result.success("Testing call. Will be removed shortly");
+      case "FirebaseModelManager#download":
+        assert (call.argument("model") != null);
+        assert (call.argument("conditions") != null);
+
+        modelName = call.argument("model");
+        Map<String, Boolean> conditionsToMap = call.argument("conditions");
+        remoteModel = new FirebaseCustomRemoteModel.Builder(modelName).build();
+        FirebaseModelDownloadConditions.Builder conditionsBuilder =
+            new FirebaseModelDownloadConditions.Builder();
+        if (conditionsToMap.get("requireCharging")) conditionsBuilder.requireCharging();
```

Comment:
@bparrishMines Just curious, for flutter channels is there a more type safe way of passing arguments and return values back (ie via protobufs). Or is it usually done with primitives like strings etc? Just curious about typical convention, and not a blocker for this PR. Thanks

Reply:
I am not sure if it fully answers your question, but

"The standard platform channels use a standard message codec that supports efficient binary serialization of simple JSON-like values, such as booleans, numbers, Strings, byte buffers, and Lists and Maps of these (see [StandardMessageCodec](https://api.flutter.dev/flutter/services/StandardMessageCodec-class.html)) for details). The serialization and deserialization of these values to and from messages happens automatically when you send and receive values."

The full guide on channels is [here](https://flutter.dev/docs/development/platform-integration/platform-channels).